pallets/subtensor/src/lib.rs

```diff
@@ -116,6 +115,15 @@ pub mod pallet {
         pub placeholder2: u8,
     }
 
+    /// Struct for NeuronCertificate.
+    pub type NeuronCertificateOf = NeuronCertificate;
+    /// Data structure for NeuronCertificate information.
+    #[derive(Decode, Encode, Default, TypeInfo, PartialEq, Eq, Clone, Debug)]
+    pub struct NeuronCertificate {
+        ///  The neuron certificate.
+        pub certificate: Vec<u8>,
```

Comment:
Can we make it a bounded vector instead? There should be a limit of how much data can an axon store on-chain.

Reply:
It is better to add freeze_struct, even for this simple struct.

Reply:
will add freeze_struct

Reply:
@andreea-popescu-reef what about the using a bounded vec ? Have you tested with actual certificate sizes to see how much data it is ?

Reply:
for an ECDSA public key we would need 256bits. To support multiple algorithms 512bits should be enough for the key plus some additional bits to determine the algoritthm so let's reserve 520bits.
I'd also rename the `certificate field` - `public_key` would be more appropiate

Reply:
Its not about the naming , but more about people being able to dump arbitrary data (potentialy 5 MB worth)  into the field. This is why we are suggesting a bounded vec. It would also be nice to see some tests showing those bounds are enforced

Reply:
made the changes